# HG changeset patch
# User Andrew Azores <aazores@redhat.com>
# Date 1374502410 14400
# Node ID 6904f82aa50185e2de370c069dcdcd2e8fe507c9
# Parent  7c75bf721d7cd342b62a2118c8de5867a60eb4f4
Added java.vm.name read permission to fix Rhino evaluation of proxy PAC (RH982558)

diff -r 7c75bf721d7c -r 6904f82aa501 netx/net/sourceforge/jnlp/runtime/RhinoBasedPacEvaluator.java
--- a/netx/net/sourceforge/jnlp/runtime/RhinoBasedPacEvaluator.java    Thu Jul 18 08:53:46 2013 +0200
+++ b/netx/net/sourceforge/jnlp/runtime/RhinoBasedPacEvaluator.java    Mon Jul 22 10:13:30 2013 -0400
@@ -48,6 +48,7 @@
 import java.security.Permissions;
 import java.security.PrivilegedAction;
 import java.security.ProtectionDomain;
+import java.util.PropertyPermission;

 import net.sourceforge.jnlp.util.TimedHashMap;

@@ -124,9 +125,15 @@

         EvaluatePacAction evaluatePacAction = new EvaluatePacAction(pacContents, pacUrl.toString(),
                 pacHelperFunctionContents, url);
+
+        // Purposefully giving only these permissions rather than using java.policy. The "evaluatePacAction"
+        // isn't supposed to do very much and so doesn't require all the default permissions given by
+        // java.policy
         Permissions p = new Permissions();
         p.add(new RuntimePermission("accessClassInPackage.org.mozilla.javascript"));
         p.add(new SocketPermission("*", "resolve"));
+        p.add(new PropertyPermission("java.vm.name", "read"));
+
         ProtectionDomain pd = new ProtectionDomain(null, p);
         AccessControlContext context = new AccessControlContext(new ProtectionDomain[] { pd });


